src/sys/ptrace/linux.rs

```diff
@@ -195,6 +195,63 @@ libc_enum! {
     }
 }
 
+#[cfg(all(
+    target_os = "linux",
+    target_env = "gnu",
+    any(
+        target_arch = "x86_64",
+        target_arch = "x86",
+        target_arch = "aarch64",
+        target_arch = "riscv64",
+    )
+))]
+/// Represents register set areas, such as general-purpose registers or floating-point registers.
+pub trait RegisterSet {
+    /// Corresponding type of registers in the kernel.
+    const VALUE: RegisterSetValue;
+
+    /// Struct representing the register space.
+    type Regs;
+}
+
+#[cfg(all(
+    target_os = "linux",
+    target_env = "gnu",
+    any(
+        target_arch = "x86_64",
+        target_arch = "x86",
+        target_arch = "aarch64",
+        target_arch = "riscv64",
+    )
+))]
+/// Register sets used in [`getregset`] and [`setregset`]
+pub mod regset {
+    use super::*;
+
+    #[derive(Debug, Clone, Copy)]
+    /// General-purpose registers.
+    pub enum NT_PRSTATUS {}
```

Comment:
Nit: maybe using a unit struct here would be more suitable?

```suggestion
    pub struct NT_PRSTATUS;
```

Reply:
Just realized that using a zero-variants enum would be better than a unit struct if we don't want our users to construct this type, so I guess this is the reason why you were using it here?

Reply:
Yes. Unit struct is OK since it should be impossible to misuse it, but enum would probably be more elegant. Up to you :)

Reply:
Yeah, I think your idea is better and we should use a zero-variants enum, so I will file a PR for this:)